### PR TITLE
Adding MinGasAmount

### LINKED ISSUE #867: EVMOS Min Global Fee
IBC transactions from Evmos to Osmosis don't seem to go through due to not being above recently implemented min global fee.

Currently using `v2.0.0-rc3`.

Error log:
```
lvl=info msg="Error building or broadcasting transaction" provider_type=cosmos chain_id=evmos_9001-2 attempt=1 max_attempts=5 error="rpc error: code = InvalidArgument desc = provided fee < minimum global fee (0aevmos < ). Please increase the gas price.: insufficient fee: invalid request"
```

`rly` config for evmos

I have tried increasing `gas-prices` and adding `gas` flag with substantial amount. Same logs occur.
```
    evmos:
        type: cosmos
        value:
            key: evmos
            chain-id: evmos_9001-2
            rpc-addr: http://localhost:16197
            account-prefix: evmos
            keyring-backend: test
            gas-adjustment: 1.5
            gas-prices: 25000000000aevmos
            debug: true
            timeout: 20s
            output-format: json
            sign-mode: direct
```            

Running EVMOS binary locally, with `~/.evmosd/config/app.toml` min-gas adjusted as well:
```
###############################################################################
###                           Base Configuration                            ###
###############################################################################

# The minimum gas prices a validator is willing to accept for processing a
# transaction. A transaction's fees must meet the minimum of any denomination
# specified in this config (e.g. 0.25token1;0.0001token2).
minimum-gas-prices = "21000000000aevmos"
```

### PR DESCRIPTION
**Pull request by joeabbey**
Fixes https://github.com/cosmos/relayer/issues/867

Enables ChainConfigs to set a MinGasAmount to workaround AnteHandlers that disallow 0 gas.

For instance on evmos, users observed:

```
lvl=info msg="Error building or broadcasting transaction" provider_type=cosmos 
chain_id=evmos_9001-2 attempt=1 max_attempts=5 error="rpc error: code = InvalidArgument 
desc = provided fee < minimum global fee (0aevmos < ). 
Please increase the gas price.: insufficient fee: invalid request"
```

Users may now set a min-gas-amount to start gas estimation with a non-zero amount.

```
    evmos:
        type: cosmos
        value:
            key: relayer
            chain-id: evmos_9001-2
            rpc-addr: http://127.0.0.1:26657
            account-prefix: evmos
            keyring-backend: test
            gas-adjustment: 1.2
            gas-prices: 20000000000aevmos
            min-gas-amount: 1
            debug: false
            timeout: 20s
            output-format: json
            sign-mode: direct
```

> **Note: min-gas-amount is not enforced**
  If a min-gas-amount of 100000 is specified and the gas estimation returns 50000 the gas adjustment factor will be applied and 60000 gas will be "wanted".

### CODE REVIEW

**Review by boojamya — APPROVED** [review]
Nice!
Would you mind just adding a note about this in the troubleshooting section?
https://github.com/cosmos/relayer/blob/main/docs/troubleshooting.md

The troubleshooting doc needs some work, but getting this in there for now will be helpful.

**Review by DavidNix — APPROVED** [review]
To echo @boojamya, it would be nice to have documentation on how this is used or what issue it solves.

**Comment by joeabbey** [comment]
Added documentation on what it solves in troubleshooting.

Included a reference to this issue in CHANGELOG.md

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -55,4 +55,33 @@ you can open `http://localhost:7597` in your browser to explore details from the
...
+            account-prefix: evmos
+            keyring-backend: test
+            gas-adjustment: 1.2
+            gas-prices: 20000000000aevmos
+            min-gas-amount: 1
+            debug: false
+            timeout: 20s
+            output-format: json
+            sign-mode: direct
+```
```
Comment (DavidNix) [inline docs/troubleshooting.md:86]: ❤️

**Review by agouin — APPROVED** [review]
:rocket: rly get rekt

**Pull request merged**